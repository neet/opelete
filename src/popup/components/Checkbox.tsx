import React from 'react';
import Toggle from 'react-toggle';

export interface Props {
  text?: string;
  value?: boolean;
  children?: React.ReactNode;
  onChange: () => void;
}

export class Checkbox extends React.PureComponent<Props> {

  public static defaultProps = {
    value: false,
  };

  public render () {
    const { text, value, children, onChange } = this.props;

    return (
      <label className='preference-modifier' >
        <div className='preference-modifier__label'>
          {children || text || ''}
        </div>

        <Toggle
          checked={value}
          icons={false}
          onChange={onChange}
        />
      </label>
    );
  }

}
