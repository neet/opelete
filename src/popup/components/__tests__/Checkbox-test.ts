import React from 'react';
import Checkbox from '../Checkbox';
import renderer from 'react-test-renderer';

describe('Checkbox', () => {
  it('renders a checkbox', () => {
    const handler = jest.fn();
    const tree = renderer.create(
      <Checkbox
        className='foo'
        value={false}
        text='hoge'
        onChange={handler}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
