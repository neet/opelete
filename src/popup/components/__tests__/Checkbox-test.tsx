import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Checkbox } from '../Checkbox';

describe('Checkbox', () => {
  it('renders a checkbox', () => {
    const handler = jest.fn();
    const tree = renderer.create(
      <Checkbox
        checked={false}
        text='hoge'
        onChange={handler}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
