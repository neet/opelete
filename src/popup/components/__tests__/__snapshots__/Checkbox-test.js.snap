// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders a checkbox 1`] = `
<label
  className="preference-modifier"
>
  <div
    className="preference-modifier__label"
  >
    hoge
  </div>
  <div
    className="react-toggle"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      />
      <div
        className="react-toggle-track-x"
      />
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      checked={false}
      className="react-toggle-screenreader-only"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      type="checkbox"
    />
  </div>
</label>
`;
